database/sqlite/sqlite_health.c

```diff
@@ -0,0 +1,1357 @@
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 31, cfg->p_db_lookup_options);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 32, cfg->p_db_lookup_after);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+
+        param++;
+        rc = sqlite3_bind_int(res, 33, cfg->p_db_lookup_before);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+    } else {
+        param++;
+        rc = sqlite3_bind_null(res, 29);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 30);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 31);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 32);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+        param++;
+        rc = sqlite3_bind_null(res, 33);
+        if (unlikely(rc != SQLITE_OK))
+            goto bind_fail;
+    }
+
+    param++;
+    rc = sqlite3_bind_int(res, 34, cfg->p_update_every);
+    if (unlikely(rc != SQLITE_OK))
+        goto bind_fail;
+
+    rc = execute_insert(res);
+    if (unlikely(rc != SQLITE_DONE))
+        error_report("Failed to store alert config, rc = %d", rc);
+
+    rc = sqlite3_reset(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to reset statement in alert hash_id store function, rc = %d", rc);
+
+    return 0;
+
+    bind_fail:
+    error_report("Failed to bind parameter %d to store alert hash_id, rc = %d", param, rc);
+    rc = sqlite3_reset(res);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Failed to reset statement in alert hash_id store function, rc = %d", rc);
+    return 1;
+}
+
+/*
+ * Select an alert config
+ */
+#define SQL_SELECT_ALERT_CONFIG_WITH_HASH "select hash_id, alarm, template, on_key, class, component, type, os, hosts, lookup, every, units, calc, families, plugin, module, green, red, warn, crit, exec, to_key, info, delay, options, repeat, host_labels, p_db_lookup_dimensions, p_db_lookup_method, p_db_lookup_options, p_db_lookup_after, p_db_lookup_before, p_update_every from alert_hash where hash_id = @hash_id;"
+#define SQL_SELECT_ALERT_CONFIG "select hash_id, alarm, template, on_key, class, component, type, os, hosts, lookup, every, units, calc, families, plugin, module, green, red, warn, crit, exec, to_key, info, delay, options, repeat, host_labels, p_db_lookup_dimensions, p_db_lookup_method, p_db_lookup_options, p_db_lookup_after, p_db_lookup_before, p_update_every from alert_hash;"
+void sql_select_alert_config(char *hash_str, BUFFER *wb)
+{
+    int rc;
+    sqlite3_stmt *res_alert = NULL;
+
+    uuid_t hash_id;
+    if (hash_str)
+        uuid_parse(hash_str, hash_id);
```

Comment:
hash_str may not be a valid uuid so uuid_parse will fail in this case

Reply:
Ok, added a check, thanks!